Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -0,0 +1,343 @@
+	if (FParse::Param(*CommandLine, *ReadFromCommandLineKey))
+	{
+		UE_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("Found ReadFromCommandLineKey in command line Keys, worker flags for custom spawning will be ignored."));
+
+		FParse::Value(*CommandLine, *TotalPlayerCommandLineKey, ExpectedPlayers);
+
+		int32 NumNPCs = 0;
+		FParse::Value(*CommandLine, *TotalNPCsCommandLineKey, NumNPCs);
+		SetTotalNPCs(NumNPCs);
+
+		FParse::Value(*CommandLine, *MaxRoundTripCommandLineKey, MaxClientRoundTripSeconds);
+		FParse::Value(*CommandLine, *MaxLatenessCommandLineKey, MaxClientViewLatenessSeconds);
+	}
+	else if (GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking())
+	{
+		USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(GetNetDriver());
+		check(NetDriver);
+
+		UE_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("Using worker flags to load custom spawning parameters."));
+		FString ExpectedPlayersString, TotalNPCsString, MaxRoundTrip, MaxViewLateness;
+
+		USpatialWorkerFlags* SpatialWorkerFlags = NetDriver != nullptr ? NetDriver->SpatialWorkerFlags : nullptr;
+		check(SpatialWorkerFlags != nullptr);
+
+		if (SpatialWorkerFlags != nullptr)
+		{
+			if (SpatialWorkerFlags->GetWorkerFlag(TotalPlayerWorkerFlag, ExpectedPlayersString))
+			{
+				ExpectedPlayers = FCString::Atoi(*ExpectedPlayersString);
+			}
+
+			if (SpatialWorkerFlags->GetWorkerFlag(TotalNPCsWorkerFlag, TotalNPCsString))
+			{
+				SetTotalNPCs(FCString::Atoi(*TotalNPCsString));
+			}
+
+			if (SpatialWorkerFlags->GetWorkerFlag(MaxRoundTripWorkerFlag, MaxRoundTrip))
+			{
+				MaxClientRoundTripSeconds = FCString::Atoi(*MaxRoundTrip);
+			}
+
+			if (SpatialWorkerFlags->GetWorkerFlag(MaxLatenessWorkerFlag, MaxViewLateness))
+			{
+				MaxClientViewLatenessSeconds = FCString::Atoi(*MaxViewLateness);
+			}
+
+			FOnWorkerFlagsUpdatedBP WorkerFlagDelegate;
+			WorkerFlagDelegate.BindDynamic(this, &ABenchmarkGymGameModeBase::OnWorkerFlagUpdated);
+			SpatialWorkerFlags->BindToOnWorkerFlagsUpdated(WorkerFlagDelegate);
+		}
+	}
+
+	UE_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("Players %d, NPCs %d, RoundTrip %d, ViewLateness %d"), ExpectedPlayers, TotalNPCs, MaxClientRoundTripSeconds, MaxClientViewLatenessSeconds);
+}
+
+void ABenchmarkGymGameModeBase::OnWorkerFlagUpdated(const FString& FlagName, const FString& FlagValue)
+{
+	if (FlagName == TotalPlayerWorkerFlag)
+	{
+		ExpectedPlayers = FCString::Atoi(*FlagValue);
+	}
+	else if (FlagName == TotalNPCsWorkerFlag)
+	{
+		SetTotalNPCs(FCString::Atoi(*FlagValue));
+	}
+	else if (FlagName == MaxRoundTripWorkerFlag)
+	{
+		MaxClientRoundTripSeconds = FCString::Atoi(*FlagValue);
+	}
+	else if (FlagName == MaxLatenessWorkerFlag)
+	{
+		MaxClientViewLatenessSeconds = FCString::Atoi(*FlagValue);
+	}
+
+	UE_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("Worker flag updated - Flag %s, Value %s"), *FlagName, *FlagValue);
+}
+
+void ABenchmarkGymGameModeBase::SetTotalNPCs(int32 Value)
+{
+	TotalNPCs = Value;
```

Comment:
Do you only want to call `OnTotalNPCsUpdated` if this value is different? There may be a disconnect otherwise as the non-auth server won't have `OnRepTotalNPCs` called on it if the values haven't changed.